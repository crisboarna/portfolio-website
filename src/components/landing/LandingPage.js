import React from 'react';
import Scrollchor from 'react-scrollchor';

import LandingPageTyping from './LandingPageTyping';
import '../../styles/landing/LandingPage.css';

import mp4Video from '../../media/LandingVideo.mp4';
import webmVideo from '../../media/LandingVideo.webm';
import ogvVideo from '../../media/LandingVideo.ogv';
import LandingVideoSnapshot from '../../media/LandingVideoSnapshot.jpg';

const LandingPage = () => {
  return (
    <div id='landingPage'>
      <LandingPageTyping/>
      <video autoPlay muted loop>
        <source src={mp4Video} type="video/mp4"/>
        <source src={webmVideo} type="video/webm"/>
        <source src={ogvVideo} type="video/ogv"/>
        <img src={LandingVideoSnapshot} alt='Video Background'/>
      </video>
      <div id='scrollButton' className='scrollDownButton'>
        <Scrollchor to='#profile'><span></span>Scroll</Scrollchor>
      </div>
    </div>
  );
};

export default LandingPage;