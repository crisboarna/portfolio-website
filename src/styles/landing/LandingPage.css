video {
    position: absolute;
    width: auto;
    height: auto;
    min-width: 100%;
    min-height: 100%;
    z-index: -5000;
}

#landingPage {
    overflow: hidden;
    width:100%;
    height: 100vh;
    position: relative;
}

.scrollDownButton a {
    position: absolute;
    bottom: 20px;
    left: 48%;
    z-index: 2000;
    display: inline-block;
    -webkit-transform: translate(0, -50%);
    transform: translate(0, -50%);
    color: #fff;
    font : normal 400 20px/1 'Josefin Sans', sans-serif;
    letter-spacing: .1em;
    text-decoration: none;
    transition: opacity .3s;
}
.scrollDownButton a:hover {
    opacity: .5;
}

div #scrollButton {
    position: relative;
    width: 100%;
    height: 100%;
}

div #scrollButton a {
    padding-top: 60px;
}

div #scrollButton a span {
    position: absolute;
    top: 0;
    left: 50%;
    width: 46px;
    height: 46px;
    margin-left: -23px;
    border: 1px solid #fff;
    border-radius: 100%;
    box-sizing: border-box;
}

div #scrollButton a span::before {
    position: absolute;
    top: 0;
    left: 0;
    z-index: 2000;
    content: '';
    width: 44px;
    height: 44px;
    box-shadow: 0 0 0 0 rgba(255,255,255,.1);
    border-radius: 100%;
    opacity: 0;
    -webkit-animation: sdb03 3s infinite;
    animation: sdb03 3s infinite;
    box-sizing: border-box;
}

div #scrollButton a span::after {
    position: absolute;
    top: 50%;
    left: 50%;
    content: '';
    width: 16px;
    height: 16px;
    margin: -12px 0 0 -8px;
    border-left: 1px solid #fff;
    border-bottom: 1px solid #fff;
    -webkit-transform: rotate(-45deg);
    transform: rotate(-45deg);
    box-sizing: border-box;
}

@-webkit-keyframes sdb03 {
    0% {
        opacity: 0;
    }
    30% {
        opacity: 1;
    }
    60% {
        box-shadow: 0 0 0 60px rgba(255,255,255,.1);
        opacity: 0;
    }
    100% {
        opacity: 0;
    }
}

@keyframes sdb03 {
    0% {
        opacity: 0;
    }
    30% {
        opacity: 1;
    }
    60% {
        box-shadow: 0 0 0 60px rgba(255,255,255,.1);
        opacity: 0;
    }
    100% {
        opacity: 0;
    }
}