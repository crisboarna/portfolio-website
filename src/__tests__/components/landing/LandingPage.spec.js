import React from 'react';
import { shallow, mount } from 'enzyme';
import LandingPage from '../../../components/landing/LandingPage';

describe('LandingPage', () => {
  it('renders without crashing', () => {
    const component = shallow(<LandingPage />);
    expect(component.exists()).toEqual(true);
  });

  it('compare to snapshot', () => {
    const component = mount(<LandingPage/>);
    expect(component).toMatchSnapshot();
  });
});