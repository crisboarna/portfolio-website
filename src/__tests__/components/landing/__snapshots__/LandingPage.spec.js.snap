// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage compare to snapshot 1`] = `
<LandingPage>
  <div
    id="landingPage"
  >
    <LandingPageTyping>
      <div
        className="row"
      >
        <Typist
          avgTypingDelay={100}
          className="col-xs-2 col-sm-offset-4 typing-top landing-typing"
          cursor={
            Object {
              "hideWhenDone": true,
              "hideWhenDoneDelay": 0,
            }
          }
          delayGenerator={[Function]}
          onCharacterTyped={[Function]}
          onLineTyped={[Function]}
          onTypingDone={[Function]}
          startDelay={0}
          stdTypingDelay={25}
        >
          <div
            className="Typist col-xs-2 col-sm-offset-4 typing-top landing-typing"
          >
            <Cursor
              blink={true}
              element="|"
              hideWhenDone={true}
              hideWhenDoneDelay={0}
              isDone={false}
              show={true}
            >
              <span
                className="Cursor Cursor--blinking"
              >
                |
              </span>
            </Cursor>
          </div>
        </Typist>
      </div>
    </LandingPageTyping>
    <video
      autoPlay={true}
      loop={true}
      muted={true}
    >
      <source
        src="LandingVideo.mp4"
        type="video/mp4"
      />
      <source
        src="LandingVideo.webm"
        type="video/webm"
      />
      <source
        src="LandingVideo.ogv"
        type="video/ogv"
      />
      <img
        alt="Video Background"
        src="LandingVideoSnapshot.jpg"
      />
    </video>
    <div
      className="scrollDownButton"
      id="scrollButton"
    >
      <Scrollchor
        to="#profile"
      >
        <a
          href="#profile"
          onClick={[Function]}
        >
          <span />
          Scroll
        </a>
      </Scrollchor>
    </div>
  </div>
</LandingPage>
`;
